### PR TITLE
chore: upgrade to prisma 5.4.2

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Upgrades prisma to 5.4.2

## Type of change

- [x] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- Checks should be passing

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by zomars** [comment]
Current dependencies on/for this PR:
* main
  * **PR #11666** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/11666" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/11666?utm_source=stack-comment).

**Review by PeerRich — DISMISSED** [review]
i dont believe i saw breaking changes from 5.2 to 5.3.1

**Review by PeerRich — DISMISSED** [review]

**Comment by zomars** [comment]
Fixed booking filters which now doesn't allow undefined values in "AND" clauses in "where" filters.

File: packages/trpc/server/routers/viewer/bookings/get.handler.ts [hunk 1]
```diff
@@ -170,12 +170,12 @@ async function getBookings({
...
-    filters &&
-    Object.keys(filters).map((key) => {
-      return bookingWhereInputFilters[key];
-    });
-
+  const filtersCombined: Prisma.BookingWhereInput[] = !filters
+    ? []
+    : Object.keys(filters)
+        .map((key) => bookingWhereInputFilters[key])
+        // On prisma 5.4.2 passing undefined to where "AND" causes an error
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/bookings/get.handler.ts:177]: Left a note here

**Review by ThyMinimalDev — APPROVED** [review]
Looks good to me 👍 

checked out the branch locally and tested ✅

**Pull request merged**